server/src/main/java/com/google/fhir/gateway/AllowedQueriesChecker.java

```diff
@@ -92,6 +92,29 @@ private boolean requestMatches(RequestDetailsReader requestDetails, AllowedQuery
 
     if (!Strings.isNullOrEmpty(entry.getRequestType())
         && !requestDetails.getRequestType().name().equalsIgnoreCase(entry.getRequestType())) {
+
+      if (!requestDetails.getRequestPath().endsWith(ProxyConstants.HTTP_URL_SEPARATOR)
+          && requestDetails.getRequestPath().contains(ProxyConstants.HTTP_URL_SEPARATOR)
```

Comment:
What is the case we are trying to avoid here? It seems to me you are trying to exclude `path` configs ending in `/ANY_VALUE` but I fail to understand why such cases should ignore requet-type (if it is provided in config)?

BTW, if we need this, the logic of this condition is complex enough to warrant a function, IMHO. Please also add unit-tests that cover this case and was not covered by the old logic.

Reply:
Reviewed and tested this, we don't need it actually - removed it